dali/base/dasds.cpp

```diff
@@ -6477,14 +6477,11 @@ void CCovenSDSManager::saveDelta(const char *path, IPropertyTree &changeTree)
     if (externalEnvironment)
     {
         // don't save any changed to /Environment if external
-        if (0 == strncmp("/Environment", path, strlen("/Environment")))
-        {
-            OWARNLOG("Attempt to change read-only Dali environment, path = %s", path);
-            return;
-        }
-        if (0 == strcmp("/", path) && changeTree.hasProp("*[@name=\"Environment\"]"))
+
+        if (startsWith(path, "/Environment") || (streq(path, "/") && changeTree.hasProp("*[@name=\"Environment\"]")))
         {
-            OWARNLOG("Attempt to change read-only Dali environment, path = %s", path);
+            querySessionManager().refreshWhiteList();
+            PROGLOG("Dali Environment updated, path = %s", path);
```

Comment:
Why drop the read-only message ?  Because the whitelist is updated now ?

Reply:
It wasn't correct before. It is valid to update the /Environment via some routes, e.g. updtdalienv - so the warning was misleading.